pkg/broker/docker_manager.go

```diff
@@ -0,0 +1,88 @@
+package broker
+
+import (
+	"context"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/cloudfoundry-incubator/blockhead/pkg/config"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/pborman/uuid"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+type dockerContainerManager struct {
+	client DockerClient
+	logger lager.Logger
+}
+
+func NewDockerContainerManager(logger lager.Logger, client DockerClient) ContainerManager {
+	return dockerContainerManager{
+		client: client,
+		logger: logger.Session("docker-container-manager"),
+	}
+}
+
+func (dc dockerContainerManager) Provision(service *config.Service, plan *config.Plan) (brokerapi.ProvisionedServiceSpec, error) {
+	ctx := context.Background()
```

Comment:
do we need to think about plumbing contexts through the entire stack? looks like we do get a context from the calling function in `broker.go`

Reply:
yeah we should probably do better plumbing of the `context`